src/Parsers/ParserCreateQuery.cpp

```diff
@@ -1510,6 +1584,9 @@ bool ParserCreateViewQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expec
         }
     }
 
+    if (!sql_security && !sql_security_p.parse(pos, sql_security, expected))
+        sql_security = std::make_shared<ASTSQLSecurity>();
```

Comment:
It looks strange that after you failed to parse SQL security you're adding a default one to the query anyway. It seems better not to do that:

```
if (!sql_security)
    sql_security_p.parse(pos, sql_security, expected);
```